theme/static/js/generic-resource.js

```diff
@@ -333,6 +333,26 @@ $(document).ready(function () {
         })
     });
 
+    $("#btn-delete-resource").on('click', function(e) {
+        e.stopImmediatePropagation();
+        // disable the delete icon to prevent users from clicking it again until it is done
+        $('#delete').addClass('disabled');
+        $('#delete').removeAttr('data-toggle');
+        $.ajax({
+            type: "POST",
+            url: "/hsapi/_internal/" + SHORT_ID + "/delete-resource/",
+            success: function (task) {
+                $('#delete-resource-dialog').modal('hide');
+                notificationsApp.registerTask(task);
+                notificationsApp.show();
+            },
+            error: function (xhr, errmsg, err) {
+                display_error_message('Failed to delete the resource', xhr.responseText);
+                $('#delete-resource-dialog').modal('hide');
+            }
+        })
+    });
+
```

Comment:
Sometimes the dialogs persist information in HS, not sure if that applies to text only or also button states. Mostly for edge cases I can't think of, might consider a `complete()` or `always()`. `always` is for jQuery 3 so I believe you would want to use the deprecated `complete` to remove the disabled class

Reply:
I don't quite follow here. In this specific ajax call case, a response will be received immediately to respond to the resource delete request . I might not get what you really meant here. By dialogs persisting, do you mean the delete resource dialog? I never run into issues with it, though.

Reply:
Dialogs persist information in other areas of HydroShare, I haven't seen it in the notifications bell. As a best practice, I'd handle re-enabling the button with a complete().

Reply:
I suppose if you already know that the dialog is being destroyed then you wouldn't need to do this, but I suspect these modals are not re-instantiated in some cases.